### ISSUE TITLE
Misleading error message when trying to save to NetCDF

### ISSUE THREAD

**Issue by rcomer** [SciTools/iris#2919]
If I have a cube and try to save it to a NetCDF file in a directory that does not exist like this:
```python
import iris

cube = iris.cube.Cube(range(7), long_name='test_cube')
iris.save(cube, 'non_dir/test.nc')
```

I get 
```
IOError: Permission denied
```

If instead I try to save it to pp file in that directory, I more correctly get
```
IOError: [Errno 2] No such file or directory: 'non_dir/test.pp'
```

**Comment by ajdawson** [comment]
The `netCDF4` module will raise the exception `OSError: Permission denied` whn you try and open a netcdf file that doesn't exist:

```python
import netCDF4

ds = netCDF4.Dataset("path/to/somewhere/imaginary/test.nc", "w")
```

I think iris is just re-issuing this error-message wrapped in an `IOError`. There probably isn't a lot iris can/should do about this, other than catching that exception, checking if it is really a permission error or a file not found, then re-issuing a possibly different exception.

It might be better to raise this issue over at [netcdf4-python](https://github.com/Unidata/netcdf4-python), see if they can change the error message.

**Comment by pelson** [comment]
I'm inclined to agree with you @ajdawson - we would like to inherit good behaviour from our dependencies (netcdf4python).

The fact that we wrap the error and change it from an OSError to an IOError is interesting too. Perhaps we should push that down to netcdf4python too!

@rcomer - are you happy to open the issue there, or would you like me to do it?

**Comment by rcomer** [comment]
@pelson would you mind raising it?  I feel like I don't entirely know what I'm talking about as I've never used that library directly.

**Comment by pelson** [comment]
Just raised in https://github.com/Unidata/netcdf4-python/issues/751.

I also just learnt that IOError is now merged into OSError for python 3.3: ``Changed in version 3.3: EnvironmentError, IOError, WindowsError, socket.error, select.error and mmap.error have been merged into OSError, and the constructor may return a subclass.``

Depending on how the issue goes upstream, I'm not against there being an explicit check in iris to provide consistent permission checking before writing to disk. Would happily review any PRs along that line.
